volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -587,6 +587,15 @@ def is_suspicious(self):
             ret = True
         elif flags_str == "r-x" and self.vm_file.dereference().vol.offset == 0:
             ret = True
+        elif "x" in flags_str:
+            for i in range(self.vm_start,self.vm_end,constants.linux.PAGE_SHIFT):
+                try:
```

Comment:
Should we keep trying all the other pages if one fails or can the exception move outside the loop?

Reply:
I changed it to stop searching in that VMA if a bad translation occurs.

Reply:
I added comments and improved logging